airflow/models/taskinstance.py

```diff
@@ -426,6 +436,10 @@ class TaskInstance(Base, LoggingMixin):
     # migration, we are keeping it as DateTime pending a change where expensive
     # migration is inevitable.
 
+    # The reason why the trigger should time out after trigger_timeout
+    # (e.g. "execution_timeout", "sensor_timeout", "trigger_timeout")
+    trigger_timeout_reason = Column(String(256))
```

Comment:
why doesn't this make more sense on the `trigger` model?

Reply:
i see there's already a trigger_timeout column on TI.

does seem to make more sense on trigger. but not sure if we should break with what's there already.  just, ti tends to be one of the larger tables, and it's already pretty wide.

Reply:
> does seem to make more sense on trigger. but not sure if we should break with what's there already. just, ti tends to be one of the larger tables, and it's already pretty wide.

it depends on if we decide to handle the timeout in the task instance as I do now, or the trigger cleanup as you propose.